Reorganize Quickstart Guide

fitzthum opened the pull request:
We've let the quickstart guide get overgrown. This PR makes it much quicker.

I moved a lot of content into separate files. I don't want to de-emphasize anything, but putting everything in one file simply does not scale and it seriously compromises the focus of what is supposed to be an approachable starting point for people coming to the project. I have tried to include pointers to the new documents in every relevant section. I don't want any of the great work people have already done on the guide to be lost.

I ditched our old four-step structure (which we weren't really following anyway) and opted for a different four-step structure, which I think is a bit simpler. Now the quickstart guide focuses on installation, basic unencrypted workloads, encryption, singing, and attestation using the generic KBS. Everything else has been moved to other files.

Hopefully I'm not the only one who thinks this is an improvement because it was a huge pain. I think this should hold us over until we get serious about having a documentation website. lmk what you think.

fitzthum committed:
Rename guides

Use snake_case and remove redundant guide name

Signed-off-by: Tobin Feldman-Fitzthum <tobin@ibm.com>

fitzthum committed:
SEV: fix heading level

Signed-off-by: Tobin Feldman-Fitzthum <tobin@ibm.com>

fitzthum committed:
SEV: Add guide overview

Signed-off-by: Tobin Feldman-Fitzthum <tobin@ibm.com>

fitzthum committed:
SEV: Add annotations guide

The shim now supports a nmber of annotations for SEV(-ES),
meaning that we no longer need to modify the config file
to set things like the guest policy or kbs uri. Update
the quickstart guide to spread the news.

Signed-off-by: Tobin Feldman-Fitzthum <tobin@ibm.com>

sameo commented on quickstart.md:
```diff
@@ -222,184 +224,40 @@ Now go back to the k8s node and ensure that you still don’t have any bitnami/n
 
-A tenant-side CoCo Key Broker System cluster include:
+The follow describes how to run and provision the generic KBS.
+The KBS should be run in a trusted environment. Thee KBS is not just one service,
```
```suggestion
The KBS should be run in a trusted environment. The KBS is not just one service,
```

sameo commented on quickstart.md:
```diff
@@ -222,184 +224,40 @@ Now go back to the k8s node and ensure that you still don’t have any bitnami/n
 
-We have prepared a sample CoCo operator custom resource that is based on the standard `ccruntime.yaml`, but in addition has the the decryption keys and configuration required to decrypt this sample container image.
-> **Note** All pods started with this sample resource will be able to decrypt the sample container and all keys shown are for demo purposes only and should not be used in production.
+To use CoCo with confidential hardware, first switch to the appropriate runtime class.
+TDX has two runtime classes, `kata-qemu-tdx` and `kata-clh-tdx`. One uses QEMU as VMM and TDVF as firmware. The other uses Cloud Hypervisor as VMM and TD-Shim as firmware.
```
I would put that part under a `### Runtime Class` section.

sameo commented on quickstart.md:
````diff
@@ -189,6 +187,10 @@ spec:
   runtimeClassName: kata
 ```
 
+Setting the runtimeclass is usually the only change needed to the pod yaml, but some platforms
````
```suggestion
Setting the `runtimeClassName` is usually the only change needed to the pod yaml, but some platforms
```

sameo reviewed (commented):
Very nice, thanks for putting this together. I only have a few nits.

dcmiddle commented on quickstart.md:
```diff
@@ -1,28 +1,40 @@
-- Running encrypted containers without CC HW (sample container images provided)
-- Running unencrypted container images or sample encrypted images with CC HW
-- Running your own encrypted container images with CC HW
+- Install CoCo using the operator.
```
extraneous .
```suggestion
- Install CoCo using the operator
```

dcmiddle commented on guides/enclave-cc.md:
```diff
@@ -0,0 +1,92 @@
+# Enclave-CC
+
```
```suggestion
This guide assumes you have already completed **Installation** and **Operator Installation** sections of the [quickstart](../quickstart.md) guide.
```

dcmiddle commented on quickstart.md:
````diff
@@ -99,6 +101,25 @@ Wait until each pod has the STATUS of Running.
+```
+for the **simulated** SGX mode build or **hardware** SGX mode build, respectively.
+
+**Note** For `enclave-cc` certain configuration changes, such as setting the
````
Delete this note if applied as suggested above.

mythi reviewed (commented):
Just a quick comment on `enclave-cc` new image URLs. Do we need similar updates for the Kata CI images too?

fitzthum committed:
Reorganize Quickstart Guide

Simplify quickstart guide to cover installation,
basic usage, encryption/signing, attestation.
Focus on the generic KBS.
Everything else is moved to other files. Pointers
to the relevant files are included where needed.

Signed-off-by: Tobin Feldman-Fitzthum <tobin@ibm.com>

fitzthum commented:
@sameo @dcmiddle PTAL at updates

dcmiddle reviewed (approved):
Looks great. Thanks for the big reorg!
